.github/workflows/autobuild.yml

```diff
@@ -1,25 +1,53 @@
 #### Automatically build and upload releases to GitHub ####
 
+
+# see analyse_git_reference.py for implementation of the logic:
+#     for every push to a branch starting with "autobuild": (can be used during development for tighter supervision of builds)
+#          - do CodeQl while building for every platform
+#          - publish the created binaries/packs only as artifacts/appendix of the github-action-run (not as release), and only retain those files for limited period
+#     for every pull-request to master:
+#          - do CodeQl while building for every platform
+#          - publish the created binaries/packs only as artifacts/appendix of the github-action-run (not as release), and only retain those files for limited period
+#     for every tag that starts with 'r_' and contains 'beta':
+#          - do CodeQl while building for every platform
+#          - publish the created binaries/packs only as artifacts/appendix as a prerelease
+#     for every tag that starts with 'r_' (and not contains 'beta'):
+#          - do CodeQl while building for every platform
+#          - publish the created binaries/packs only as artifacts/appendix as a release
+#     for  the  tag that is named "latest":
+#          - do CodeQl while building for every platform
+#          - publish the created binaries/packs only as artifacts/appendix as the release "latest"
+#
+
+
 on:
   workflow_dispatch:
   push:
     tags:
       - "r*"
       - "latest"
-    #branches:
+    branches:
```

Comment:
I think I'd still prefer `*` for the CodeQL.

If it's possible to exclude branches from the publish, then perhaps have branches for that (`dev-*` or whatever) that are automatically rejected for pull requests.

Reply:
Yes. That's a great idea!

Reply:
Do I get the idea correctly?
- branch not starting with "dev": CodeQL + artifacts (push or pull-request)
- branch starting with "dev": push: no action at all
- branch starting with "dev": pull request: Error

Reply:
Yes.